test/pipe.cc

```diff
@@ -0,0 +1,96 @@
+#include "eventuals/pipe.h"
+
+#include <thread>
+
+#include "eventuals/loop.h"
+#include "eventuals/terminal.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+
+using eventuals::Loop;
+using eventuals::Pipe;
+
+TEST(Pipe, UniqueValue) {
+  Pipe<int> pipe;
+
+  *pipe.Write(1);
+  auto e = [&pipe]() {
+    return pipe.Read()
+        | Loop<int>()
+              .context(/*sum = 0*/ 0)
+              .body([](auto&& sum, auto& repeated, auto&& value) {
+                sum += value;
+                repeated.Done();
+              })
+              .ended([](auto& sum, auto& k) {
+                k.Start(std::move(sum));
+              });
+  };
+
+  EXPECT_EQ(1, pipe.values_size());
+  EXPECT_EQ(1, *e());
+}
+
+
+TEST(Pipe, Values) {
+  Pipe<int> pipe;
+
+  auto do_write = [](Pipe<int>& pipe) {
+    for (size_t i = 1; i <= 5; ++i) {
+      *pipe.Write(i);
+    }
+  };
+
+  std::thread t{do_write, std::ref(pipe)};
```

Comment:
It's easier to read inline lambdas and to capture by ref in the lambda vs via `std::ref`.
```suggestion
  std::thread t([&pipe]() {
    for (size_t i = 1; i <= 5; ++i) {
      *pipe.Write(i);
    }
  });
```